### PR TITLE
Ability to set inner instance

### PR DESCRIPTION
**Pull request by duart38**
As it stand you can't adjust the inner state object directly by assignment (after getInstance). there's no reason this shouldn't be possible.

```TypeScript
singleton.getInstance() = {..}; // error
singleton.setInstance(...); // works
```

### CODE REVIEW

**Review by grevend — CHANGES_REQUESTED** [review]
Thanks for your contribution. Before I can merge your pull request, it would be important to add the necessary test cases and update the documentation.

File: test.ts [hunk 1]
```diff
@@ -6,3 +6,10 @@ Deno.test("check single instance creation", () => {
   const value = rand.getInstance();
   assertEquals(value, rand.getInstance());
 });
+
+Deno.test("check setting inner instance", () => {
+  const str = singleton(()=>"a");
```
Comment (grevend) [inline test.ts]: Please fix the formatting and update the documentation.

**Review by grevend — CHANGES_REQUESTED** [review]

File: singleton.bundle.js [hunk 2]
```diff
@@ -2,6 +2,8 @@ function singleton(factory) {
     let instance = null;
     return {
         getInstance: ()=>instance == null ? instance = factory() : instance
+        ,
```
Comment (grevend) [inline singleton.bundle.js:5]: Why is the `,` symbol not in the previous line?

**Review by grevend — CHANGES_REQUESTED** [review]
Please also update the documentation (README).

File: singleton.bundle.js (hunk 2 above)
Comment (duart38) [reply singleton.bundle.js:5]: Man the deno bundler is just outright ridiculous

**Pull request merged**